src/l3.c

```diff
@@ -77,20 +84,48 @@ L3_STATIC_ASSERT(sizeof(loc_t) == sizeof(uint32_t),
                  "Expected sizeof(loc_t) == 4 bytes.");
 #endif  // L3_LOC_ENABLED
 
+/**
+ * Definitions for L3_LOG.flags field. This field is used to reliably identify
+ * the provenance of a L3-log file so that the appropriate tool / technique can
+ * be used to unpack log-entries when dumping the contents of a L3-log file.
+ */
+typedef uint16_t l3_log_flags_t;
+enum l3_log_flags_t
+{
+      L3_LOG_FLAGS_LINUX                = ((uint16_t) 0x0001)
+    , L3_LOG_FLAGS_MACOSX               = ((uint16_t) 0x0002)
+
+    , L3_LOG_FLAGS_LOC_ENCODING         = ((uint16_t) 0x0010)
+    , L3_LOG_FLAGS_LOC_ELF_ENCODING     = ((uint16_t) 0x0020)
+
+};
+
 /**
  * L3 Log Structure definitions:
  */
 typedef struct l3_log
 {
-    uint64_t idx;
-    uint64_t fbase_addr;
-    uint64_t pad0;
-    uint64_t pad1;
-    L3_ENTRY slots[L3_MAX_SLOTS];
+    uint64_t        idx;
+    uint64_t        fbase_addr;
+    uint32_t        pad0;
+    uint16_t        pad2;
+    l3_log_flags_t  flags;
+    uint64_t        pad1;
```

Comment:
Sizeof the struct has not changed. Just externalized some pad fields into `magic` (L110) and `flags` (L112).

Reply:
Use `__attribute__((align(n)))`?

Reply:
I'd like to stay with the scheme of externalizing pad-bytes added for compiler-padding as, then, it's apparent that some spare bytes are available for future development.

Also, I'm a bit leery of relying on compiler `__attribute__`s -- just in case those are not supported similarly on all compilers (e.g. clang, MS-Visual C++ etc.)